public/app/features/explore/utils/supplementaryQueries.test.ts

```diff
@@ -145,17 +144,24 @@ jest.mock('@grafana/runtime', () => ({
   },
 }));
 
-const setup = async (rootDataSource: string, type: SupplementaryQueryType, targetSources?: string[]) => {
-  const rootDataSourceApiMock = await getDataSourceSrv().get({ uid: rootDataSource });
-
-  targetSources = targetSources || [rootDataSource];
+const setup = async (targetSources: string[], type: SupplementaryQueryType) => {
+  const datasources: Array<{ datasource: DataSourceApi; targets: DataQuery[] }> = [];
 
   const requestMock = new MockDataQueryRequest({
     targets: targetSources.map((source, i) => new MockQuery(`${i}`, 'a', { uid: source })),
   });
   const explorePanelDataMock: Observable<ExplorePanelData> = mockExploreDataWithLogs();
 
-  return getSupplementaryQueryProvider(rootDataSourceApiMock, type, requestMock, explorePanelDataMock);
+  for (const i in targetSources) {
+    const source = targetSources[i];
+    const datasource = await getDataSourceSrv().get({ uid: source });
+    datasources.push({
+      datasource,
+      targets: [new MockQuery(`${i}`, 'a', { uid: source })],
+    });
+  }
```

Comment:
you should be able to do this concurrently with something like:

```ts
const datasources = await Promise.all(
    targetSources.map(async (source, i) => {
      const datasource = await getDataSourceSrv().get({ uid: source });
      return {
        datasource,
        targets: [new MockQuery(`${i}`, 'a', { uid: source })],
      };
    })
  );
```

Reply:
Fixed in c6339c147e017355ae093e2d4ba059de62608f24